### DISCUSSION TITLE
Pull config.yml from Git hub (or other source)

### DISCUSSION THREAD

**Discussion by lethalox** [0xERR0R/blocky#1016] [Ideas]
I would like to store my config.yml in a GitHub (self hosted or not).   Since I run block on two machines with the same config.yml.   It would be nice to point both at the GitHub and have them pull config on start.   There could be cache version local, if for some reason the git or url was not responding.

**Comment by ThinkChaos** [comment]
Both this and #1015 sound to me like you should look into some devops setup.  
There's a bunch of ways you can get a `git push` to trigger a deploy on your infrastructure, which would include updating the config.

**Comment by buroa** [comment]
I do it in my homelab: https://github.com/buroa/k8s-gitops

https://github.com/buroa/k8s-gitops/tree/master/kubernetes/apps/networking/blocky/app

**Reply by lethalox** [reply]
Where in your config to pull from github?

**Comment by lethalox** [comment]
Thanks for sharing.   I don't use kubernetes (yet) so I am trying to translate this back to docker-compose files.

**Reply by shizunge** [reply]
I think we can use docker config to share configuration files between replicates of services.

https://docs.docker.com/compose/compose-file/compose-file-v3/#configs-configuration-reference

**Comment by sn0opy** [comment]
Had a similar issue when I added a fallback server to my home network. In my case, I simply build my own blocky image via Github actions and publish the image via Github's container registry. Said image has my config already included. The repo itself is private and so is the image, but this is basically what I use: https://gist.github.com/sn0opy/9472fff1c119a424c591b9068e80c421
